primitives/npos-elections/compact/src/lib.rs

```diff
@@ -256,6 +258,29 @@ fn struct_def(
 				Ok(assignments)
 			}
 		}
+		type __IndexAssignment = _npos::IndexAssignment<
+			<#ident as _npos::CompactSolution>::Voter,
+			<#ident as _npos::CompactSolution>::Target,
+			<#ident as _npos::CompactSolution>::Accuracy,
+		>;
+		impl<'a> _npos::sp_std::convert::TryFrom<&'a [__IndexAssignment]> for #ident {
```

Comment:
looks correct, but a test for this conversion is the final verdict.

Reply:
Test added in 73d3fdd6c.